openfeature/provider/__init__.py

```diff
@@ -17,6 +21,20 @@ class ProviderStatus(Enum):
     STALE = "STALE"
     FATAL = "FATAL"
 
+    __events__: typing.ClassVar[typing.Dict[ProviderEvent, str]] = {
+        ProviderEvent.PROVIDER_READY: READY,
+        ProviderEvent.PROVIDER_ERROR: ERROR,
+        ProviderEvent.PROVIDER_ERROR: FATAL,
+        ProviderEvent.PROVIDER_STALE: STALE,
+    }
+
+    @classmethod
+    def from_provider_event(
+        cls, event: ProviderEvent
+    ) -> typing.Optional[ProviderStatus]:
+        value = ProviderStatus.__events__.get(event)
+        return ProviderStatus[value] if value else None
```

Comment:
```suggestion
        return ProviderStatus.__events__.get(event)
```
I don't think you need to invoke `ProviderStatus` again and `.get()` anyway returns `None` if it can't find what you are looking for 🙂

Reply:
This was done because the dict values are strings when the dict is defined. In any case, I decided to inline the code to avoid exposing an internal detail